addons/mail/models/ir_attachment.py

```diff
@@ -10,6 +10,15 @@
 class IrAttachment(models.Model):
     _inherit = 'ir.attachment'
 
+    @api.model
+    def check(self, mode, values=None):
+        super().check(mode, values=values)
+        if mode != 'unlink' or not self or self.env.is_admin():
```

Comment:
I can still write in rpc on the attachment (and change the file eg), shouldn't we also check for write ?